managed/services/management/agent_test.go

```diff
@@ -0,0 +1,183 @@
+// Copyright (C) 2017 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package management
+
+import (
+	"context"
+	"testing"
+
+	"github.com/AlekSi/pointer"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/reform.v1"
+	"gopkg.in/reform.v1/dialects/postgresql"
+
+	"github.com/percona/pmm/api/managementpb"
+	"github.com/percona/pmm/managed/models"
+	"github.com/percona/pmm/managed/utils/logger"
+	"github.com/percona/pmm/managed/utils/testdb"
+	"github.com/percona/pmm/managed/utils/tests"
+)
+
+func TestAgentService(t *testing.T) {
+	setup := func(t *testing.T) (context.Context, *AgentService, func(t *testing.T), *mockPrometheusService) { //nolint:unparam
+		t.Helper()
+
+		ctx := logger.Set(context.Background(), t.Name())
+		uuid.SetRand(&tests.IDReader{})
+
+		sqlDB := testdb.Open(t, models.SetupFixtures, nil)
+		db := reform.NewDB(sqlDB, postgresql.Dialect, reform.NewPrintfLogger(t.Logf))
+
+		vmdb := &mockPrometheusService{}
+		vmdb.Test(t)
+
+		state := &mockAgentsStateUpdater{}
+		state.Test(t)
+
+		ar := &mockAgentsRegistry{}
+		ar.Test(t)
+
+		teardown := func(t *testing.T) {
```

Comment:
I think instead of retruning teardown function we can add required items to `t.Cleanup(...)` right here.

Reply:
We can do that, but IMO it would require to repeat lines 55 thru 60 of the code found in `teardown` everywhere, no? Or am I thinking it wrong?